nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -531,9 +549,15 @@ in {
     services.postfix = lib.mkIf (cfg.smtp.createLocally && cfg.smtp.host == "127.0.0.1") {
       enable = true;
     };
-    services.redis = lib.mkIf (cfg.redis.createLocally && cfg.redis.host == "127.0.0.1") {
-      enable = true;
-    };
+    services.redis = lib.mkMerge [
+      (lib.mkIf (cfg.redis.createLocally && cfg.redis.host == "127.0.0.1") {
+        enable = true;
+      })
+      (lib.mkIf cfg.redis.enableUnixSocket {
+        unixSocket = "/run/redis/redis.sock";
+        unixSocketPerm = 770;
```

Comment:
this options does not really exist according to the CI.
```
The option `services.redis.unixSocketPerm' does not exist. Definition values:
```

This can only be set via https://search.nixos.org/options?channel=unstable&show=services.redis.settings&from=0&size=50&sort=relevance&query=services.redis.

Reply:
This option is added to PR - https://github.com/NixOS/nixpkgs/pull/117456

Reply:
Still 
```
error: The option `services.redis.unixSocketPerm' does not exist. Definition values:
- In `/home/runner/work/nixpkgs/nixpkgs/nixos/modules/services/web-apps/mastodon.nix':
    {
      _type = "if";
      condition = false;
      content = {
        _type = "if";
    ...
(use '--show-trace' to show detailed location information)
Error: Process completed with exit code 1.
```

Reply:
Fixed